lib/Optimizer/Transforms/LoopUnroll.cpp

```diff
@@ -184,6 +184,33 @@ class LoopUnrollPass : public cudaq::opt::impl::LoopUnrollBase<LoopUnrollPass> {
       emitError(UnknownLoc::get(ctx), "did not unroll loops");
       signalPassFailure();
     }
+
+    // First get a count of the number of times a variable is used.
+    std::unordered_map<std::string, int> myCounts;
+    op->walk([&](mlir::Operation *walkOp) {
```

Comment:
We really need a `quake::MeasurementInterface`, I've asked this before I think, do we have one ? @schweitzpgi 

e.g.
```cpp
op->walk([&](quake::MeasurementInterface measureOp) { ... });
```

Reply:
I don't think this has anything to do with loop unrolling.

It looks like an independent pass that just happens to be placed here.

Reply:
The reason I placed it here is because this is the section of code that expands a single `mz(qubit array)` and unrolls it into a bunch of single `mz(qubit)` calls, and since I was wanting unique registerName values for each unrolled iteration, I thought it would make sense to put it here.

Reply:
I updated the code to return a `WalkResult`. Let me know if you'd like to see any additional changes here.

Reply:
I do agree with Eric's point, and might ask him to further comment on it. Should this not be in a separate rewriter, one that we add to the unrolling-pipeline?

Reply:
I discussed this with Eric and have submitted 454a687e to refactor some of the changes. It pulls my changes to the `LoopUnrollPass` class and places them in a new pass that I added to the `unrolling-pipeline`.